source/transactions-convenient-api/tests/commit-writeconcernerror.yml

```diff
@@ -0,0 +1,196 @@
+          errorCodeName: UnknownReplWriteConcern
+          errorLabelsOmit: ["TransientTransactionError", "UnknownTransactionCommitResult"]
+    expectations:
+      -
+        command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - { _id: 1 }
+            ordered: true
+            lsid: session0
+            txnNumber: { $numberLong: "1" }
+            startTransaction: true
+            autocommit: false
+            # omitted fields
+            readConcern: ~
+            writeConcern: ~
+          command_name: insert
+          database_name: *database_name
+      -
+        command_started_event:
+          command:
+            commitTransaction: 1
+            lsid: session0
+            txnNumber: { $numberLong: "1" }
+            autocommit: false
+            # omitted fields
+            readConcern: ~
+            startTransaction: ~
+            writeConcern: ~
+          command_name: commitTransaction
+          database_name: admin
+    outcome:
+      collection:
+        data: []
+  -
+    description: commitTransaction is not retried after write concern error (UnsatisfiableWriteConcern)
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { times: 1 }
+      data:
+          failCommands: ["commitTransaction"]
+          closeConnection: false
+          writeConcernError:
+            code: 100
+            codeName: UnsatisfiableWriteConcern
+            errmsg: "Not enough data-bearing nodes"
+    operations:
+      -
+        name: withTransaction
+        object: session0
+        callback:
+          operations:
+            -
+              name: insertOne
+              object: collection
+              arguments:
+                session: session0
+                document: { _id: 1 }
+              result:
+                insertedId: 1
+        result:
+          errorCodeName: UnsatisfiableWriteConcern
+          errorLabelsOmit: ["TransientTransactionError", "UnknownTransactionCommitResult"]
+    expectations:
+      -
+        command_started_event:
+          command:
+            insert: *collection_name
+            documents:
+              - { _id: 1 }
+            ordered: true
+            lsid: session0
+            txnNumber: { $numberLong: "1" }
+            startTransaction: true
+            autocommit: false
+            # omitted fields
+            readConcern: ~
+            writeConcern: ~
+          command_name: insert
+          database_name: *database_name
+      -
+        command_started_event:
+          command:
+            commitTransaction: 1
+            lsid: session0
+            txnNumber: { $numberLong: "1" }
+            autocommit: false
+            # omitted fields
+            readConcern: ~
+            startTransaction: ~
+            writeConcern: ~
+          command_name: commitTransaction
+          database_name: admin
+    outcome:
+      collection:
+        data: []
```

Comment:
This expectation is not being satisfied. It seems that the server commits the transaction even though it responds with a writeConcernError.

Not sure if there's a way to indicate in the failpoint that the commit shouldn't succeed.

Reply:
This seems like the same issue you raised in https://github.com/mongodb/specifications/pull/435#discussion_r245158138. See my responses there and acknowlege if the commit resolves things.

Based on the original transaction spec tests, write ops are still applied after write concern error returned by a fail point, so I think it's correct for all of the `commit-writeconcernerror.yml` tests to expect a single `{ _id: 1 }` document in their outcome data.